ui/packages/catalog-ui-search/src/main/webapp/js/model/Query.js

```diff
@@ -157,15 +207,31 @@ Query.Model = PartialAssociatedModel.extend({
   },
   initialize: function() {
     this.currentIndexForSource = {}
+    this.state = [{}]
 
     _.bindAll.apply(_, [this].concat(_.functions(this))) // underscore bindAll does not take array arg
     this.set('id', this.getId())
-    this.listenTo(
-      user.get('user>preferences'),
-      'change:resultCount',
-      this.handleChangeResultCount
-    )
     this.listenTo(this, 'change:cql', () => this.set('isOutdated', true))
+
+    const sync = () => {
+      this.dispatch(updateResults(this.get('result').toJSON()))
+      this.set('serverPageIndex', serverPageIndex(this.state))
+
+      const totalHits = this.get('result')
+        .get('status')
+        .reduce((total, status) => {
```

Comment:
✏️ can you do `total = 0` instead of the initial 0 value?

Reply:
That's generally not how reduce functions are written.